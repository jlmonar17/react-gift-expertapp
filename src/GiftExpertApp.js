import { useState } from "react";

const GiftExpertApp = () => {
    const [categories, setCategories] = useState([
        "One Punch",
        "Naruto",
        "Dragon Ball",
    ]);

    const handleAdd = () => {
        // setCategories([...categories, "Ranma"]);
        setCategories((cats) => [...cats, "Ranma 1/2"]);
    };

    return (
        <>
            <h2>GiftExpertApp</h2>
            <hr />

            <button onClick={handleAdd}>Agregar</button>

            <ul>
                {categories.map((category) => {
                    return <li key={category}> {category}</li>;
                })}
            </ul>
        </>
    );
};

export default GiftExpertApp;
