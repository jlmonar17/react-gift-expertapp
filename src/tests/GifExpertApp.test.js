import { shallow } from "enzyme";
import GiftExpertApp from "../GiftExpertApp";

describe("Pruebas para <GiftExpertApp />", () => {
    test("Prueba que GiftExpertApp se renderice correctamente", () => {
        const wrapper = shallow(<GiftExpertApp />);

        expect(wrapper).toMatchSnapshot();
    });

    test("Debe de mostrarse ua lista de categorias", () => {
        const categories = ["AC/DC", "Metallica"];
        const wrapper = shallow(
            <GiftExpertApp defaultCategories={categories} />
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("GifGrid").length).toBe(categories.length);
    });
});
