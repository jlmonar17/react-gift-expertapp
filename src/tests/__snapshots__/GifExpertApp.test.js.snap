// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para <GiftExpertApp /> Debe de mostrarse ua lista de categorias 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="AC/DC"
    key="AC/DC"
  />
  <GifGrid
    category="Metallica"
    key="Metallica"
  />
</Fragment>
`;

exports[`Pruebas para <GiftExpertApp /> Prueba que GiftExpertApp se renderice correctamente 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;
